src/main/java/uk/ac/ebi/eva/commons/models/converters/data/DBObjectToSamplesConverter.java

```diff
@@ -0,0 +1,145 @@
+     *
+     * @param sampleIds Map of samples to sampleId
+     */
+    public DBObjectToSamplesConverter(boolean compressDefaultGenotype, Map<String, Integer> sampleIds) {
+        this(compressDefaultGenotype);
+        setSampleIds(sampleIds);
+    }
+
+
+    @Override
+    public VariantSourceEntry convert(DBObject object) {
+        
+        if (sampleIds == null || sampleIds.isEmpty()) {
+            return new VariantSourceEntry(object.get(FILEID_FIELD).toString(), object.get(STUDYID_FIELD).toString());
+        }
+        
+        BasicDBObject mongoGenotypes = (BasicDBObject) object.get(SAMPLES_FIELD);
+//        int numSamples = idSamples.size();
+        
+        // Temporary file, just to store the samples
+        VariantSourceEntry fileWithSamples = new VariantSourceEntry(object.get(FILEID_FIELD).toString(), 
+                object.get(STUDYID_FIELD).toString());
+
+        // Add the samples to the file
+        for (Map.Entry<String, Integer> entry : sampleIds.entrySet()) {
+            Map<String, String> sampleData = new HashMap<>(1);  //size == 1, only will contain GT field
+            fileWithSamples.addSampleData(entry.getKey(), sampleData);
+        }
+
+        // An array of genotypes is initialized with the most common one
+        String mostCommonGtString = mongoGenotypes.getString("def");
+        if(mostCommonGtString != null) {
+            for (Map.Entry<String, Map<String, String>> entry : fileWithSamples.getSamplesData().entrySet()) {
+                entry.getValue().put("GT", mostCommonGtString);
+            }
+        }
+
+        // Loop through the non-most commmon genotypes, and set their value
+        // in the position specified in the array, such as:
+        // "0|1" : [ 41, 311, 342, 358, 881, 898, 903 ]
+        // genotypes[41], genotypes[311], etc, will be set to "0|1"
+        for (Map.Entry<String, Object> dbo : mongoGenotypes.entrySet()) {
+            if (!dbo.getKey().equals("def")) {
+                String genotype = dbo.getKey().replace("-1", ".");
+                for (Integer sampleId : (List<Integer>) dbo.getValue()) {
+                    if (idSamples.containsKey(sampleId)) {
+                        fileWithSamples.getSamplesData().get(idSamples.get(sampleId)).put("GT", genotype);
+                    }
+                }
+            }
+        }
+
+
+        
+        return fileWithSamples;
+    }
+
+
+    public void setSamples(List<String> samples) {
+        int i = 0;
+        int size = samples == null? 0 : samples.size();
+        sampleIds = new HashMap<>(size);
+        idSamples = new HashMap<>(size);
+        if (samples != null) {
+            for (String sample : samples) {
+                sampleIds.put(sample, i);
+                idSamples.put(i, sample);
+                i++;
+            }
+        }
+    }
+
+    public void setSampleIds(Map<String, Integer> sampleIds) {
+        this.sampleIds = sampleIds;
```

Comment:
That is dangerous, any change to the structure that is passed to this function will modify it also.